sebs/knative/knative.py

```diff
@@ -0,0 +1,552 @@
+
+                    # Create the function object
+                    res = KnativeFunction(
+                        sanitize_benchmark_name,
+                        code_package.benchmark,
+                        code_package.hash,
+                        function_cfg,
+                    )
+
+                    # Add HTTP trigger with the function URL
+                    trigger = LibraryTrigger(
+                        sanitize_benchmark_name, self.get_knative_func_cmd()
+                    )
+                    trigger.logging_handlers = self.logging_handlers
+                    res.add_trigger(trigger)
+
+                    return res
+
+                except subprocess.CalledProcessError as e:
+                    self.logging.error(
+                        f"Error deploying Knative function {sanitize_benchmark_name}."
+                    )
+                    self.logging.error(f"Output: {e.stderr.decode('utf-8')}")
+                    raise RuntimeError(e) from e
+
+        except FileNotFoundError:
+            self.logging.error(
+                "Could not retrieve Knative functions - is path to func correct?"
+            )
+            raise RuntimeError("Failed to access func binary")
+
+    def update_function(self, function: Function, code_package: Benchmark):
+        self.logging.info(f"Updating an existing Knative function {function.name}.")
+        function = cast(KnativeFunction, function)
+        docker_image = self.system_config.benchmark_image_name(
+            self.name(),
+            code_package.benchmark,
+            code_package.language_name,
+            code_package.language_version,
+        )
+
+        try:
+            subprocess.run(
+                [
+                    *self.get_knative_func_cmd(),
+                    "deploy",
+                    "--path",
+                    code_package.code_location,
+                ],
+                capture_output=True,
+                check=True,
+            )
+            function.config.docker_image = docker_image
+
+        except FileNotFoundError as e:
+            self.logging.error(
+                "Could not update Knative function - is the 'func' CLI installed and configured correctly?"
+            )
+            raise RuntimeError(e) from e
+        except subprocess.CalledProcessError as e:
+            self.logging.error(f"Unknown error when running function update: {e}!")
+            self.logging.error(
+                "Ensure the SeBS cache is cleared if there are issues with Knative!"
+            )
+            self.logging.error(f"Output: {e.stderr.decode('utf-8')}")
+            raise RuntimeError(e)
+
+    def update_function_configuration(
+        self, function: Function, code_package: Benchmark
+    ):
+        self.logging.info(f"Updating an existing Knative function {function.name}.")
+        docker_image = self.system_config.benchmark_image_name(
```

Comment:
Where do you update the following properties:
- Memory allocation
- Timeout
- Environment variables?